test/src/test/java/io/seata/xa/XAModeTest2.java

```diff
@@ -0,0 +1,600 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.xa;
+
+import com.alibaba.druid.pool.DruidDataSource;
+import com.alibaba.druid.pool.xa.DruidXADataSource;
+import com.alibaba.druid.util.JdbcUtils;
+import com.mysql.jdbc.jdbc2.optional.MysqlXADataSource;
+import io.seata.core.context.RootContext;
+import io.seata.core.exception.TransactionException;
+import io.seata.core.model.BranchStatus;
+import io.seata.core.model.BranchType;
+import io.seata.core.model.Resource;
+import io.seata.rm.DefaultResourceManager;
+import io.seata.rm.datasource.xa.AbstractDataSourceProxyXA;
+import io.seata.rm.datasource.xa.DataSourceProxyXA;
+import io.seata.rm.datasource.xa.DataSourceProxyXANative;
+import io.seata.rm.datasource.xa.ResourceManagerXA;
+import io.seata.rm.datasource.xa.XAXid;
+import io.seata.rm.datasource.xa.XAXidBuilder;
+import io.seata.spring.annotation.GlobalTransactionScanner;
+import io.seata.tm.api.GlobalTransaction;
+import io.seata.tm.api.GlobalTransactionContext;
+import oracle.jdbc.xa.client.OracleXADataSource;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.postgresql.xa.PGXADataSource;
+
+import javax.sql.DataSource;
+import javax.sql.XAConnection;
+import javax.sql.XADataSource;
+import javax.transaction.xa.XAException;
+import javax.transaction.xa.XAResource;
+import javax.transaction.xa.Xid;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class XAModeTest2 {
+
+    private static final int testRecordId = 888;
+    private static final String testRecordName = "xxx";
+    private static final long testTid = 1582688600006L;
+    private static final String mockXid = "127.0.0.1:8091:" + testTid;
+    private static final long mockBranchId = testTid + 1;
+
+    private static final String pg_jdbcUrl = "jdbc:postgresql://127.0.0.1:5432/postgres";
```

Comment:
Unable to connect to 127.0.0.1 database, CI occurred a large number of error logs. Please use mock or disable, and remove all printf logs.